tests/ha/single_pod_restart/test_data_pod_restart.py

```diff
@@ -647,47 +607,110 @@ def test_mpu_after_pod_restart(self):
         obj_size = result[1]["ContentLength"]
         LOGGER.debug("Uploaded object info for %s is %s", self.bucket_name, result)
         assert_utils.assert_equal(obj_size, file_size * const.Sizes.MB)
-        upload_checksum = str(resp[2])
-        LOGGER.info("Step 5: Sucessfully performed multipart upload for  size %s MB in "
+        upload_checksum_1 = str(resp[2])
+        LOGGER.info("Step 1: Successfully performed multipart upload for size %s MB in total %s "
+                    "parts.", file_size, total_parts)
+
+        LOGGER.info("Step 2: Shutdown random data pod by deleting deployment and "
+                    "verify cluster & remaining pods status")
+        resp = self.ha_obj.delete_kpod_with_shutdown_methods(
+            master_node_obj=self.node_master_list[0], health_obj=self.hlth_master_list[0],
+            down_method=const.RESTORE_DEPLOYMENT_K8S)
+        # Assert if empty dictionary
+        assert_utils.assert_true(resp[1], "Failed to shutdown/delete pod")
+        pod_name = list(resp[1].keys())[0]
+        self.deployment_name = resp[1][pod_name]['deployment_name']
+        self.deployment_backup = resp[1][pod_name]['deployment_backup']
+        self.restore_method = resp[1][pod_name]['method']
+        assert_utils.assert_true(resp[0], "Cluster/Services status is not as expected")
+        LOGGER.info("Step 2: Successfully shutdown data pod %s. Verified cluster and "
+                    "services states are as expected & remaining pods status is online.", pod_name)
+
+        LOGGER.info("Step 3: Download the uploaded object %s in Step 1 & verify checksum",
+                    self.object_name)
+        resp = s3_test_obj.object_download(self.bucket_name, self.object_name, download_path_1)
+        LOGGER.info("Download object response: %s", resp)
+        assert_utils.assert_true(resp[0], resp[1])
+        download_checksum_1 = self.ha_obj.cal_compare_checksum(file_list=[download_path_1],
+                                                               compare=False)[0]
+        assert_utils.assert_equal(upload_checksum_1, download_checksum_1,
+                                  f"Failed to match checksum: {upload_checksum_1},"
+                                  f" {download_checksum_1}")
+        LOGGER.info("Matched checksum: %s, %s", upload_checksum_1, download_checksum_1)
+        LOGGER.info("Step 3: Successfully downloaded the object %s & verified the checksum",
+                    self.object_name)
+
+        download_path_2 = os.path.join(self.test_dir_path, self.test_file + "_download_2")
```

Comment:
why not define these before actual steps started? Any reason to shift them here?

Reply:
No specific reason, I can move variable declaration at the top of the test.